lib/dry/monads/result.rb

```diff
@@ -107,11 +106,9 @@ def to_s
         end
         alias_method :inspect, :to_s
 
-        # @return [Maybe::Some]
         def to_maybe
-          Kernel.warn 'Success(nil) transformed to None' if @value.nil?
-          Dry::Monads::Maybe(@value)
-        end
+          raise "Load Maybe first with require 'dry/monads/maybe'"
+        end unless method_defined?(:to_maybe)
```

Comment:
@flash-gordon You happy with the way I'm conditionally defining these methods? If `"dry/monads/maybe"` is required first I don't want it's helpful implementation of `Result::Success#to_maybe` to be overridden by this one here.